anda/lib/directxshadercompiler/DirectXShaderCompiler.spec

```diff
@@ -0,0 +1,243 @@
+to compile High-Level Shader Language (HLSL) programs into DirectX Intermediate
+Language (DXIL) representation. Applications that make use of DirectX for
+graphics, games, and computation can use it to generate shader programs.
+
+%description devel
+This package contains library and header files needed to develop new native
+programs that use %{name}.
+
+%description libs
+This package contains runtime shared libraries needed to link with native
+programs that use %{name}.
+
+Provided files include:
+* libdxcompiler.so, a .so providing a componentized compiler, assembler,
+disassembler, and validator.
+* and some other ones!
+
+%description tools
+This package contains runtime binaries of %{name} that could be used in
+scripts.
+
+Provided files include:
+* dxc, a command-line tool that can compile HLSL programs for SM6.0 or later.
+* dxv, a command-line tool for validating the DXIL bytecode/compiled binaries.
+* and some other ones!
+
+%description cmake-utils
+This package contains CMake scripts of %{name}.
+Theoretically internal usage only, likely no usage at all.
+
+%description static
+Static libraries / binaries to link with applications at compile-time
+for %{name}.
+
+# Prepare.
+ 
+%prep
+%autosetup
+# Are we building with external libraries?
+%if %{with external_libraries}
+    # If so, extract and prepare them.
+    %setup -D -C -a 1 -q
+    %setup -D -C -a 2 -q
+    %setup -D -C -a 3 -q
+    rm -rf external/DirectX-Headers
+    rm -rf external/SPIRV-Headers
+    rm -rf external/SPIRV-Tools
+    mv DirectX-Headers-%{DirectX_Headers_HASH} external/DirectX-Headers
+    mv SPIRV-Headers-%{SPIRV_Headers_HASH} external/SPIRV-Headers
+    mv SPIRV-Tools-%{SPIRV_Tools_HASH} external/SPIRV-Tools
+%else
+    # Warn :P.
+    %{warn: Building without external libraries is unsupported by upstream!}
+    %{warn: You are likely to run into compilation failures this way.}
+    # Otherwise, nuke external libraries as a whole.
+    rm -rf external/
+%endif
+ 
+# Build.
+# Attribution: https://github.com/gentoo/guru/blob/master/dev-util/DirectXShaderCompiler/DirectXShaderCompiler-1.8.2407.ebuild
+%build
+%cmake \
+    -C ./cmake/caches/PredefinedParams.cmake \
+    -DSPIRV_WERROR=0 \
+    -DLLVM_BUILD_DOCS=0 \
+    -DLLVM_BUILD_TOOLS=0 \
+    -DSPIRV_BUILD_TESTS=0 \
+    -DLLVM_ENABLE_WERROR=0 \
+    -DBUILD_SHARED_LIBS=OFF \
+    -DLLVM_VERSION_SUFFIX=dxc \
+    -DSPIRV_WARN_EVERYTHING=0 \
+    -DCMAKE_INSTALL_PREFIX="%{_libdir}/%{name}" \
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo
+%cmake_build
+ 
+# Install.
+
+%install
+%cmake_install
+mkdir %{buildroot}%{_bindir} -p
+# Link binaries.
+for BINARY in dxa dxc dxl dxopt dxr dxv; do
+ln -s ../%{lib}/%{name}/bin/$BINARY %{buildroot}%{_bindir}/$BINARY;
+done
+
+# Test.
+%check
+%ctest
+ 
+# Files.
+%files tools
+%{_libdir}/%{name}/bin/
+%{_bindir}/dxa
+%{_bindir}/dxc
+%{_bindir}/dxl
+%{_bindir}/dxopt
+%{_bindir}/dxr
+%{_bindir}/dxv
+
```

Comment:
why putting everything into the subpackage when you can just put them into the main package?